src/diffusions.jl

```diff
@@ -165,17 +164,18 @@ Example
 
 ~~~~
 """
-function pagerank_power!{T}(x::Vector{T}, y::Vector{T},
+function pagerank_power!(x::Vector{T}, y::Vector{T},
     P, alpha::T, v, tol::T,
-    maxiter::Int, iterfunc::Function)
-    ialpha = 1./(1.-alpha)
+    maxiter::Int, iterfunc::Function) where T
+    ialpha = 1. /(1. -alpha)
     xinit = x
     _applyv!(x,v,0.,1.) # iteration number 0
     iterfunc(0,x)
     for iter=1:maxiter
-        # y = P*x
-        A_mul_B!(y,P,x)
-        gamma = 1.-alpha*sum_kbn(y)
+        y = P*x
```

Comment:
This is wrong isn't it? Won't this allocate a new vector?

Reply:
Yes, my mistake -- the correct line was right below it (`mul!(y,P,x)`). Fixed now 2488e36adc7a947be8cba864c5d175c95263bf12